jubako/src/main/java/com/justeat/jubako/JubakoAdapter.kt

```diff
@@ -118,17 +53,20 @@ open class JubakoAdapter(
         return holder
     }
 
-    override fun onBindViewHolder(holder: JubakoViewHolder<Any>, position: Int) {
-        logger.log(TAG, "Bind ViewHolder")
-
-        val item = data.getItem(position)
-
-        holder.onClickDelegate = { postViewHolderEvent(Event.Click(item.id, it)) }
-
-        attachReloader(item, holder)
+    override fun getItemCountActual(): Int = data.numItemsLoaded()
+    override fun getCurrentState(): PaginatedDataState<*> = state
+    override fun loadMore() = load()
+    override fun hasMoreToLoad(): Boolean {
+        return hasMore
+    }
 
-        bindWhenNewOrDefault(item, holder)
+    override fun getItem(position: Int): Any = data.getItem(position)
 
+    override fun bindItemToHolder(holder: JubakoViewHolder<Any>, item: Any) {
+        val description = (item as ContentDescription<Any>)
```

Comment:
Why not defining in the function param as `item: ContentDescription<Any>` then ?

Reply:
Very good catch @Electryc was an oversight since the generic data type was specified as `Any` and not `ContentDescription<Any>`!